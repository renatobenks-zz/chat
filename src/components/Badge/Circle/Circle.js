// @flow

import * as React from 'react';
import styled, { css } from 'styled-components';

const SVGCircle = styled.circle`
  fill: ${props => props.color || props.theme.palette.white};
  ${props =>
    props.border &&
    css`
      stroke: ${props.border !== true ? props.border : props.theme.palette.black};
      stroke-width: 10px;
    `};
`;

type Props = {
  color?: string,
  border?: string,
};

const Circle = ({ color, border }: Props) => {
  return (
    <svg viewBox="0 0 120 120">
      <SVGCircle color={color} border={border} cx="60" cy="60" r="50" />
    </svg>
  );
};

export default Circle;
