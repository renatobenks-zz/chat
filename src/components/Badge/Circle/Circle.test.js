import React from 'react';

import { render } from '../../../../test/helpers';

import Circle from './Circle';

const renderer = component => render(component, { withTheme: true });

it('default circle', () => {
  expect(renderer(<Circle />)).toMatchSnapshot();
});

it('circle with custom color', () => {
  expect(renderer(<Circle color="gray" />)).toMatchSnapshot();
});

it('circle with border', () => {
  expect(renderer(<Circle border />)).toMatchSnapshot();
});

it('circle with custom border', () => {
  expect(renderer(<Circle border="red" />)).toMatchSnapshot();
});
