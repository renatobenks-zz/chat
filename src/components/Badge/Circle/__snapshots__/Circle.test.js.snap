// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle with border 1`] = `
<svg
  viewBox="0 0 120 120"
>
  <circle
    className="sc-bdVaJa giUmLf"
    cx="60"
    cy="60"
    r="50"
  />
</svg>
`;

exports[`circle with custom border 1`] = `
<svg
  viewBox="0 0 120 120"
>
  <circle
    className="sc-bdVaJa kKawRB"
    cx="60"
    cy="60"
    r="50"
  />
</svg>
`;

exports[`circle with custom color 1`] = `
<svg
  viewBox="0 0 120 120"
>
  <circle
    className="sc-bdVaJa hySWXy"
    color="gray"
    cx="60"
    cy="60"
    r="50"
  />
</svg>
`;

exports[`default circle 1`] = `
<svg
  viewBox="0 0 120 120"
>
  <circle
    className="sc-bdVaJa kShjcB"
    cx="60"
    cy="60"
    r="50"
  />
</svg>
`;
